tests/python/test_multidevice.py

```diff
@@ -112,3 +114,81 @@ def multidevice_schedule(self):
     torch.testing.assert_close(
         out_tensors[0], expected_out_tensor, rtol=1.3e-6, atol=1e-4
     )
+
+
+@pytest.mark.skipif(
+    utils.is_pre_ampere(),
+    reason="Flash Attention is only supported on Ampere and newer devices.",
+)
+@pytest.mark.mpi
+def test_sdpa(mpi_test):
+    d, b, s, a, h = mpi_test.size, 2, 1024, 12, 768
+
+    if a % d != 0:
+        pytest.skip(f"We only support even split, so {a} has to be divisible by {d}.")
+
+    class Model(FusionDefinition):
+        def definition(self) -> None:
+            self.q = self.define_tensor(
+                shape=[d, -1, -1, -1, -1],
+                contiguity=[True if d > 1 else None, True, True, True, True],
```

Comment:
why do we need this conditional here?

Reply:
https://github.com/NVIDIA/Fuser/blob/61a77e0a64d5bc446ba1c009f04a19204a28eab2/csrc/ir/nodes.cpp#L2922-L2925 would fail without this. 

However, I agree this makes `define_tensor` a bit cumbersome to use. I guess this hasn't been too annoying because `define_tensor` is usually called by Thunder-generated code and/or called with a constant shape. @jjsjann123 What do you think about this?

Reply:
Yeah this is indeed unwieldy. I think we can add another API replacing `std::vector<std::optional<bool>> contiguity` with `bool default_contiguous`.

Alternatively, currently behavior is that `if contiguity.empty()`, we default to have everything as non-contiguous. We can change this to have everything as contiguous and you wouldn't need to provide a contiguity arg to this function. 

But I'm not a big fan of altering existing behavior. Leaning towards adding another API.